evap/staff/importers.py

```diff
@@ -574,55 +574,45 @@ def write_enrollments_to_db(self, semester, vote_start_datetime, vote_end_date):
                 evaluation.participants.add(*participants)
             self.create_success_messages(students_created, responsibles_created)
 
+    @staticmethod
+    def make_phrase(count, singular, plural):
+        count_word = pgettext("count", "no") if count == 0 else str(count)
+        return f"{count_word} {ngettext(singular, plural, count)}"
+
     def create_success_messages(self, students_created, responsibles_created):
-        student_count = len(students_created)
-        student_word = pgettext("count", "no") if student_count == 0 else str(student_count)
-        responsible_count = len(responsibles_created)
-        responsible_word = pgettext("count", "no") if responsible_count == 0 else str(responsible_count)
-        evaluation_count = len(self.evaluations)
-        evaluation_word = pgettext("count", "no") if evaluation_count == 0 else str(evaluation_count)
-
-        student_phrase = ngettext("{count} student", "{count} students", student_count).format(count=student_word)
-        responsible_phrase = ngettext("{count} responsible", "{count} responsibles", responsible_count).format(
-            count=responsible_word
-        )
-        evaluation_phrase = ngettext(
-            "{count} course/evaluation", "{count} courses/evaluations", evaluation_count
-        ).format(count=evaluation_word)
+        if not students_created and not responsibles_created and not self.evaluations:
+            return _("Nothing changed")
+
+        student_phrase = self.make_phrase(len(students_created), "student", "students")
+        responsible_phrase = self.make_phrase(len(responsibles_created), "responsible", "responsibles")
+        evaluation_phrase = self.make_phrase(len(self.evaluations), "course/evaluation", "courses/evaluations")
 
         message = _("Successfully created {evaluation_phrase}, {student_phrase} and {responsible_phrase}").format(
             responsible_phrase=responsible_phrase, student_phrase=student_phrase, evaluation_phrase=evaluation_phrase
         )
-        if student_count == 0 and responsible_count == 0 and evaluation_count == 0:
-            message = "Nothing changend"
-        elif student_count > 0 or responsible_count > 0:
+
+        if students_created or responsibles_created:
             message += ":"
             message += create_user_list_html_string_for_message(students_created)
             message += create_user_list_html_string_for_message(responsibles_created)
```

Comment:
This discards the knowledge that the value from `create_user_list_html_string_for_message` is already escaped. The part inside the `if` should probably be one `message = format_html(...)`